src/tablet/tablet_impl.cc

```diff
@@ -4366,7 +4608,7 @@ void TabletImpl::SendIndexData(RpcController* controller, const ::openmldb::api:
         MemTable* mem_table = dynamic_cast<MemTable*>(table.get());
         if (mem_table == NULL) {
             PDLOG(WARNING, "table is not memtable. tid %u, pid %u", request->tid(), request->pid());
-            response->set_code(::openmldb::base::ReturnCode::kTableTypeMismatch);
+            response->set_code(::openmldb::base::ReturnCode::kOperatorNotSupport);
```

Comment:
check the `storage_mode` before doing the `dynamic_cast`. 
If `storage_mode` is not `kMemory`, `kOperationNotSupport`. Otherwise, `kTableTypeMismatch`